ngraph/core/shape_inference/include/depth_to_space_shape_inference.hpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cstdint>
+#include <openvino/core/validation_util.hpp>
+#include <openvino/op/depth_to_space.hpp>
+#include <openvino/opsets/opset1.hpp>
+
+#include "utils.hpp"
+namespace ov {
+namespace op {
+namespace v0 {
+
+template <class T>
+void shape_infer(const ov::op::v0::DepthToSpace* op,
+                 const std::vector<T>& input_shapes,
+                 std::vector<T>& output_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 1 && output_shapes.size() == 1);
+
+    const auto& data_shape = input_shapes[0];
+    const ov::Rank data_rank = data_shape.rank();
+    if (data_rank.is_static()) {
+        NODE_VALIDATION_CHECK(op,
+                              !(data_shape.size() < 3),
+                              "The input tensor with rank lower than 3 is not supported (input rank: ",
+                              data_shape.size(),
+                              ")");
+    }
+
+    if (data_shape.is_static()) {
```

Comment:
This is too strict constrain. Lets require static rank only for that code block. Also previous comments are applicable here

Reply:
Move pieces of code out but still leave some check in this strict constrain branch.  Need some operator % and operator / for the dimension element.